Resources/public/js/views/ez-usermenuview.js

```diff
@@ -0,0 +1,224 @@
+     *
+     * @module ez-usermenuview
+     */
+    Y.namespace('eZ');
+
+    var CLASS_HIDDEN = 'is-user-menu-hidden',
+        CLASS_USER_PROFILE = 'ez-user-profile',
+        SELECTOR_USER_PROFILE = '.' + CLASS_USER_PROFILE,
+        SELECTOR_USER_MENU = '.ez-user-menu';
+
+    /**
+     * The User Menu view
+     *
+     * @namespace eZ
+     * @class UserMenuView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.UserMenuView = Y.Base.create('userMenuView', Y.eZ.TemplateBasedView, [], {
+        events: {
+            '.ez-user-menu-item': {
+                'tap': '_toggleUserMenu'
+            },
+            '.ez-user-menu-item-logout': {
+                'tap': '_logOut'
+            }
+        },
+
+        initializer: function () {
+            this.on('*:toggleUserMenu', this._toggleUserMenu);
+        },
+
+        /**
+        * Renders the user menu view
+        *
+        * @method render
+        * @return {eZ.UserMenuView} the view itself
+        */
+        render: function () {
+            var container = this.get('container');
+
+            container.setHTML(this.template(this.getAttrs()));
+            container.on('clickoutside', this._hideMenuOnClickOutside, this);
+
+            this._renderMenuItems();
+
+            return this;
+        },
+
+        /**
+         * Tap event handler on the logout link
+         *
+         * @method _logOut
+         * @protected
+         * @param event {Object} tap event facade
+         */
+        _logOut: function (event) {
+            event.preventDefault();
+            this.fire('logOut', {
+                originalEvent: event
+            });
+        },
+
+        /**
+         * Toggle the user menu. 
+         *
+         * @method _toggleUserMenu
+         * @protected
+         */
+        _toggleUserMenu: function () {
+            var methodName = this.get('displayed') ? 'addClass' : 'removeClass',
+                userMenu = this.get('container').one(SELECTOR_USER_MENU);
+
+            userMenu[methodName](CLASS_HIDDEN);
+            this.set('displayed', !this.get('displayed'));
+        },
+
+        /**
+         * Checks if clicked outside the user profile and hide the user menu if is vissible. 
+         *
+         * @method _hideMenuOnClickOutside
+         * @protected
+         */
+        _hideMenuOnClickOutside: function (event) {
+            if (!event.target.hasClass(CLASS_USER_PROFILE) && event.target.ancestor(SELECTOR_USER_PROFILE) === null && this.get('displayed')) {
+                 this._toggleUserMenu();
+            }
+        },
+
+        /**
+         * Renders menu items views
+         *
+         * @method _renderMenuItems
+         * @protected
+         */
+        _renderMenuItems: function () {
+            this.get('items').forEach(Y.bind(function (view) {
+                view.addTarget(this);
+                this.get('container').one(SELECTOR_USER_MENU).append(view.render().get('container'));
```

Comment:
I would cache this in variable: `var menuItemsContainer = this.get('container').one(SELECTOR_USER_MENU);`